processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -472,7 +472,10 @@ private void sortPropertyMappingsByDependencies() {
         }
 
         private Type getReturnTypeToConstructFromSelectionParameters(SelectionParameters selectionParams) {
-            if ( selectionParams != null && selectionParams.getResultType() != null ) {
+            // resultType only applies to method that actually has @BeanMapping annotation, never to forged methods
+            if ( !( method instanceof ForgedMethod )
```

Comment:
This makes sense. However, doesn't it make more sense to construct the appropriate `MappingMethodOptions` when we create a new `ForgedMethod` instead?

Reply:
I (of course :smile: looked on that option). But
1. Its too much hassle
2. There might be actually a reason to carry down all options through the underlying forged methods.. You just have to be careful to use them and check whether they are really applicable

Reply:
We had it like that some refactorings ago..